File: Leetcode/1448 Count Good Nodes in Binary Tree.py
class Solution:
    def goodNodes(self, root, max_path_val=float("-inf")):
        if root is None:
            return 0
        max_path_val = max(max_path_val, root.val)
        n_good_nodes = int(max_path_val == root.val)
        n_good_nodes += self.goodNodes(root.left, max_path_val)
        n_good_nodes += self.goodNodes(root.right, max_path_val)
        return n_good_nodes

File: Leetcode/1448 Count Good Nodes in Binary Tree.ts
function goodNodes(root: TreeNode | null, maxNodeInPath: number = -Infinity): number {
    if (root === null) {
        return 0;
    }
    maxNodeInPath = Math.max(maxNodeInPath, root.val)
    let nGoodNodes = root.val === maxNodeInPath? 1 : 0;
    nGoodNodes += goodNodes(root.left, maxNodeInPath);
    nGoodNodes += goodNodes(root.right, maxNodeInPath);
    return nGoodNodes;
}
